### DISCUSSION TITLE
What am I creating with these templates?

### DISCUSSION THREAD

**Discussion by EmrecanKaracayir** [microsoft/WindowsAppSDK#1971] [Q&A] [answered]
UWP is clear yes. But when I create a project with **Blank App, Packaged (WinUI 3 Desktop)** template, I don't know which platform it is. Is it WPF, is it WinForms? What exactly is used in this template?

![image](https://user-images.githubusercontent.com/84916904/149169016-3e438f50-7632-451d-8672-c7a80fd4bf66.png)

**Comment by ghost1372** [comment, marked as answer]
The templates are clear, they all create a new project based on WinUI3. (with c++ or c# language)
There is no template for WPF or Winform

**Reply by EmrecanKaracayir** [reply]
So, WinUI is in the same category with WPF and Winforms. It is the next app development platform not just UI framework, am I right?

**Reply by ghost1372** [reply]
> So, WinUI is in the same category with WPF and Winforms. It is the next app development platform not just UI framework, am I right?

They are in the same category because they all use the **desktop** tag. I think Yes, you are right.

> The Windows App SDK empowers all Windows desktop apps with modern Windows UI, APIs, and platform features, including back-compat support, shipped via NuGet.

**Reply by andrewleader** [reply]
@EmrecanKaracayir thanks for the questions (and thanks ghost1372 for helping), just to add some additional clarity...

WinUI 3, WPF, and WinForms are all **UI frameworks**. They all run on the **desktop platform** (as in they have access to Win32 APIs + UWP APIs, unlike the UWP platform which only had access to UWP APIs).

Developers usually refer to their **app type** based on their predominant UI framework, so developers will often say they built a "WinUI 3 app" or a "WPF app", but both of those are running on the desktop platform and share similar platform APIs like the System.IO.File APIs, and only differ in their UI framework.

WinUI 3 is our newest and recommended UI framework choice, but you can still build a WPF or WinForms app and use the latest non-WinUI WinAppSDK APIs like the Windowing APIs and the AppLifecycle APIs. And in the future, you'll be able to add WinUI 3 controls to a WPF or WinForms app via XAML Islands.

**Reply by EmrecanKaracayir** [reply]
That made my life a lot easier. Thank you for the clear explanation.
